internal/databroker/server.go

```diff
@@ -288,31 +296,37 @@ func (srv *Server) SyncLatest(req *databroker.SyncLatestRequest, stream databrok
 	})
 }
 
-func (srv *Server) getBackend(lock bool) (backend storage.Backend, version uint64, err error) {
+func (srv *Server) getBackend() (backend storage.Backend, version uint64, err error) {
+	backend = srv.backend
+	version = srv.version
+	if backend == nil {
+		var err error
+		backend, err = srv.newBackend()
+		srv.backend = backend
+		if err != nil {
+			return nil, 0, err
+		}
+	}
+	return backend, version, nil
+}
+
+func (srv *Server) getBackendLocked() (backend storage.Backend, version uint64, err error) {
```

Comment:
the semantics I was referring to is different - the `Locked` is a method that is executed when mutex is already acquired